src/ui/token.js

```diff
@@ -51,13 +49,15 @@ class Token extends React.Component {
     x: PropTypes.number,
     y: PropTypes.number,
     z: PropTypes.number,
+    template: PropTypes.func,
+    center: PropTypes.object,
```

Comment:
I don't think we want users to manipulate `center` directly. Let's rename it to `_center` and make it an internal prop.

Reply:
The code now passes a function to recalculate graphical properties that are dependent on x, y, z, and the cellSize. Otherwise, the current animation logic does not work since it sets the coordinates on Token, and thus Token must be able to recalculate e.g. the center.